odoo/addons/base/tests/test_res_users.py

```diff
@@ -217,3 +310,100 @@ def test_reified_groups_on_change(self):
+                UserForm._values[self.sales_categ_field] = self.group_sales_user.id
+                UserForm._perform_onchange([self.sales_categ_field])
+
+                self.assertEqual(
+                    UserForm.user_group_warning,
+                    'Since Test Group User is a/an Field Service Administrator, you can set Sales right only to Administrator'
+                )
+
+                UserForm._values[self.sales_categ_field] = self.group_sales_administrator.id
+                UserForm._perform_onchange([self.sales_categ_field])
+                self.assertFalse(UserForm.user_group_warning)
+
+    def test_user_multi_group_inheritance_warning(self):
+        """ User changes 'Sales: User' from 'Sales: Administrator' and
+        'Project: User' from 'Project: Administrator'. The warning should
+        be there since 'Sales: Administrator' and 'Project: Administrator'
+        are required when user is havning 'Field Service: Administrator'.
+        When user reverts the changes For 'Sales: Administrator', warning
+        should disappear for Sales Access."""
+        with self.assertQueryCount(__system__=104):
+            with Form(self.test_group_user.with_context(show_user_group_warning=True), view='base.view_users_form') as UserForm:
+                UserForm._values[self.sales_categ_field] = self.group_sales_user.id
+                UserForm._values[self.project_categ_field] = self.group_project_user.id
+                UserForm._perform_onchange([self.sales_categ_field])
+
+                warnings = [
+                    'Since Test Group User is a/an Field Service Administrator, you can set Sales right only to Administrator',
+                    'Since Test Group User is a/an Field Service Administrator, you can set Project right only to Administrator',
+                ]
+                self.assertTrue(all(warning in UserForm.user_group_warning for warning in warnings))
+
+                UserForm._values[self.sales_categ_field] = self.group_sales_administrator.id
+                UserForm._perform_onchange([self.sales_categ_field])
+
+                self.assertEqual(
+                    UserForm.user_group_warning,
+                    'Since Test Group User is a/an Field Service Administrator, you can set Project right only to Administrator'
+                )
+
+    def test_user_multi_possible_group_inheritance_warning(self):
+        """ User changes 'Timesheets: User: own timesheets only ' from
+        'Timesheets: Administrator'. The warning should be there since
+        'Timesheets: Administrator' or 'Timesheets: User: all timesheets'
+        is required when user is havning 'Project: Administrator'.
+        When user reverts the changes For 'Timesheets: User: all timesheets',
+        warning should disappear."""
+        with self.assertQueryCount(__system__=98):
+            with Form(self.test_group_user.with_context(show_user_group_warning=True), view='base.view_users_form') as UserForm:
```

Comment:
the 2 with can be merged
https://stackoverflow.com/questions/893333/multiple-variables-in-a-with-statement